src/Aeson.purs

```diff
@@ -112,24 +163,153 @@ getField' decoder obj str =
     (lmap (AtKey str) <<< decoder)
     (FO.lookup str obj)
 
+-- | Returns an Aeson available under a sequence of keys in given Aeson.
+--   If not possible returns JsonDecodeError.
+getNestedAeson :: Aeson -> Array String -> Either JsonDecodeError Aeson
+getNestedAeson asn@(Aeson { numberIndex, patchedJson: AesonPatchedJson pjson }) keys =
+  note (UnexpectedValue $ toStringifiedNumbersJson asn) $
+    mkAeson <$> (foldM lookup pjson keys :: Maybe Json)
+  where
+  lookup :: Json -> String -> Maybe Json
+  lookup j lbl = caseJsonObject Nothing (FO.lookup lbl) j
+
+  mkAeson :: Json -> Aeson
+  mkAeson json = Aeson { numberIndex, patchedJson: AesonPatchedJson json }
+
 infix 7 getField as .:
```

Comment:
I'd move this operator definition closer to the `getField` function

Reply:
Will do